collection-renderer-lib/src/cmr/collection_renderer/services/collection_renderer.clj

```diff
@@ -110,20 +112,40 @@
   "URL of the Earthdata Search application"
   {:default "https://search.earthdata.nasa.gov/search"})
 
+(defn- get-granule-count-for-collection
+  "Retrieve granule count for the collection corresponding to concept-id."
+  [context concept-id]
+  (->> {:page-size 0
+        :collection-concept-id concept-id}
+       (params/parse-parameter-query context :granule)
+       (search/find-concepts context :granule)
+       :hits))
+
+(defn- get-additional-information
+  "Retrieve additional information for collection rendering."
+  [context concept-id]
+  {"granule_count" (get-granule-count-for-collection context concept-id)})
+
 (defn render-collection
   "Renders a UMM-C collection record and returns the HTML as a string."
-  [context collection concept-id]
-  (let [umm-json (umm-json/umm->json
-                  (vm/migrate-umm context
-                                  :collection
-                                  umm-version/current-collection-version
-                                  (context->preview-gem-umm-version context)
-                                  collection))]
-
+  ([context collection concept-id]
+   (render-collection
+    context
+    collection
+    concept-id
+    (get-additional-information context concept-id)))
+  ([context collection concept-id additional-information]
```

Comment:
Question for Chris, Leo, or anyone else who has been around for the lifetime of the project:
* How likely is it that we may want to pass something else here, in addition to additional information?

If the answer to that is "probably never", then ignore the rest of this comment :-) 

I'd probably change `additional-information` to `opts`. Then, in the first arity, pass `{:additional-info {"granule..." ...}}`. Then, in `render-erb`, do the usual `"additional_information" (:additional-info opts)`. 

This makes it easy to add something new. 

Since I don't think anything else calls this, that's also a chance that could be made easily in the future, with zero impact.

Reply:
My 2 cents: `additional-information` is already a map, we will just add more entries into `additional-information` if we want to add more info in the future. The preview gem code is responsible for how to use the additional information it gets from CMR.